### PR TITLE
Implement Base and joints dynamics in TSID

### PR DESCRIPTION
**Pull request by GiulioRomualdi**

### CODE REVIEW

**Comment by GiulioRomualdi** [comment]
Friendly ping @S-Dafarra

File: src/TSID/src/JointsDynamicsTask.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+#include <iDynTree/Model/Model.h>
+
+using namespace BipedalLocomotion::ParametersHandler;
+using namespace BipedalLocomotion::System;
+using namespace BipedalLocomotion::TSID;
+
+bool JointsDynamicsTask::initialize(std::weak_ptr<ParametersHandler::IParametersHandler> paramHandler,
+                                    const System::VariablesHandler& variablesHandler)
+{
+    constexpr std::string_view errorPrefix = "[BaseDynamicsTask::initialize] ";
```
Comment (S-Dafarra) [inline src/TSID/src/JointsDynamicsTask.cpp:20]: Copypasta

File: src/TSID/src/JointsDynamicsTask.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    iDynTree::toEigen(this->subA(m_jointsTorqueVariable)).setIdentity();
+
+    return true;
+}
+
+bool JointsDynamicsTask::update()
+{
+    if (!m_kinDyn->generalizedBiasForces(m_generalizedBiasForces))
+    {
+        std::cerr << "[BaseDynamicsTask::update] Unable to get the bias forces." << std::endl;
```
Comment (S-Dafarra) [inline src/TSID/src/JointsDynamicsTask.cpp:150]: Copypasta

File: src/TSID/src/JointsDynamicsTask.cpp [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        return false;
+    }
+
+    if (!m_kinDyn->getFreeFloatingMassMatrix(m_massMatrix))
+    {
+        std::cerr << "[BaseDynamicsTask::update] Unable to get the mass matrix." << std::endl;
```
Comment (S-Dafarra) [inline src/TSID/src/JointsDynamicsTask.cpp:156]: Copypasta

File: src/TSID/src/JointsDynamicsTask.cpp [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+    m_b = m_generalizedBiasForces.tail(m_kinDyn->getNrOfDegreesOfFreedom());
+    iDynTree::toEigen(this->subA(m_robotAccelerationVariable))
+        = -m_massMatrix.bottomRows(m_kinDyn->getNrOfDegreesOfFreedom());
+
+    for (const auto& contactWrench : m_contactWrenches)
+    {
+        if (!m_kinDyn->getFrameFreeFloatingJacobian(contactWrench.frameIndex, m_jacobian))
+        {
+            std::cerr << "[BaseDynamicsTask::update] Unable to get contact wrench associated to "
```
Comment (S-Dafarra) [inline src/TSID/src/JointsDynamicsTask.cpp:168]: Copypasta

**Comment by GiulioRomualdi** [comment]
I Fixed all the typos in https://github.com/dic-iit/bipedal-locomotion-framework/pull/178/commits/021563d7ad2277f660cf77a684e41c75d9528be5

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**